Doc/library/abc.rst

```diff
@@ -129,14 +130,14 @@ a helper class :class:`ABC` to alternatively define ABCs through inheritance:
                   yield None
 
           def get_iterator(self):
-              return self.__iter__()
+              return iter(self)
```

Comment:
The existing implementation here makes it explicit that we're calling the abstract method declared just above. In the new implementation, this is obfuscated.

Reply:
Hm, sorry - I complete forgot what I was doing here 5 months ago :)
So, I will look at this PR with new state of mind.

This example is created to show how `__subclasshook__` works.

```
For a demonstration of these concepts, look at this example ABC definition::
```

But, right now it does not use anything special. Basically, you can just remove it (in the current codebase it would be a correct thing to do).

I think that we need to add `get_iterator` to the mix of `__subclasshook__` to show that classes are considered a part of `MyIterable` when they either have `__iter__` or `get_iterator`.


Plus, the docs also say:

```
The :meth:`get_iterator` method
   is also part of the ``MyIterable`` abstract base class, but it does not have
   to be overridden in non-abstract derived classes.
```

But, it actually is not in this example. Because we use `MyIterable.register(Foo)`. So, `Foo` must have this method.